go/vt/vttablet/tabletserver/messager/message_manager.go

```diff
@@ -949,3 +946,17 @@ func (mm *messageManager) readPending(ctx context.Context, bindVars map[string]*
 	}
 	return qr, err
 }
+
+// This grants the caller exclusive access to the message service.
+// When this is needed for a function, you must get the mutexes in
+// stream mutex, main mutex order to avoid deadlocks with other places
+// that conditionally get the mutexes in this same order.
+func (mm *messageManager) getExclusiveLock() {
+	mm.streamMu.Lock()
+	mm.mu.Lock()
```

Comment:
Don't we need to change the order here? `unsubscribe()` will acquire `mu` first and then `streamMu` if it needs to stop the vstream.

Reply:
Yes! Good catch :-) Corrected

Reply:
Should we add a comment in the code explaining why the ordering? If you got it backwards after diving in deep, it seems likely that we'll have no idea in a few months. :)